### ISSUE TITLE
Binary MessageAttribute is corrupted when routed to SQS Queue

### ISSUE THREAD

**Issue by gruff4l0** [localstack/localstack#2432]
# Type of request: This is a ...
[x ] bug report
# Detailed description

When a message gets posted directly to SQS Queue, then Listener is able to read payload as well as deserialize binary message attribute. 

If the same message is posted to SNS Topic first, and then gets routed to SQS Queue, 
binary message attribute becomes corrupted and not deserialize by the listener anymore. 

Both scenarios work as expected (message arrives intact and deserialize) when the same infrastructure is deployed to AWS.

## Expected behavior

Regardless, whether the message with binary message attribute is posted to SNS Topic or SQS Queue, it should arrive as it was sent, including binary message attribute. 

## Actual behavior

When a message is sent to SNS Topic first, and then routed via subscription with RawMessageDelivery to SQS Queue, the binary message attribute arrives corrupted.

# Steps to reproduce

Please see example project and readme on [gruff4l0/localstack-sns-sqs-bug-report](https://github.com/gruff4l0/localstack-sns-sqs-bug-report)

**Comment by whummer** [comment]
Thanks for reporting and preparing the sample project @gruff4l0 . A simple integration test has also been added in #2512 . We haven't been able to reproduce this issue. I was able to deploy and run your example. When running the curl requests, the following logs are printed in the console:
```
23:00:10.327 [pool-1-thread-21] INFO  localstack.pub.PubController - Pub sqs Hello
23:00:10.328 [pool-1-thread-21] DEBUG localstack.pub.PubController - sent binary is rO0ABXNyACBsb2NhbHN0YWNrLnNoYXJlZC5UcmFuc2Zlck9iamVjdHA1pSOtVRSlAgABTAAHY29udGVudHQAEkxqYXZhL2xhbmcvU3RyaW5nO3hwdAAFSGVsbG8=
23:03:28.675 [pool-1-thread-22] INFO  localstack.pub.PubController - Pub sns Hello
23:03:28.675 [pool-1-thread-22] DEBUG localstack.pub.PubController - sent binary is rO0ABXNyACBsb2NhbHN0YWNrLnNoYXJlZC5UcmFuc2Zlck9iamVjdHA1pSOtVRSlAgABTAAHY29udGVudHQAEkxqYXZhL2xhbmcvU3RyaW5nO3hwdAAFSGVsbG8=
```
... and the messages can also be retrieved from the queue:
```
$ awslocal sqs receive-message --queue-url http://localhost:4576/queue/QUEUE-1
{
    "Messages": [
        {
            "MessageId": "0f441b69-00cf-4110-b5d0-e58e866d8625",
            "ReceiptHandle": "0f441b69-00cf-4110-b5d0-e58e866d8625#0afd32b9-7da0-40b7-9766-584ab5bd18f9",
            "MD5OfBody": "dcc61e5a04e06326ab95f828edb6343b",
            "Body": "{\"size\":92,\"content\":\"Hello\"}"
        }
    ]
}
```

Can you please provide some more details about which particular error message you're seeing? Is it possible that this issue is potentially already resolved in the `latest` version of the Docker image?

**Comment by gruff4l0** [comment]
Thank you @whummer for the response. 
Publishing of message works fine when the actual error happens when message received.

To verify the error with AWS CLI supply `--message-attribute-names cargo` to see the attached binary message attribute content for attribute `cargo`. Also, it's good to get several messages at to see the difference. 

Please pull updated version of demo project (it has lates AWS SDK, better logging, toggle to disable consumer, change application port to 9000)

Steps to reproduce with AWS CLI:

**Demo App (with sqs listener disabled)**: 
$ `mvn clean package`
$ `java -jar -Dapp.consumer.enabled=false target/localstack-sns-sql-raw-message-0.2.jar`

**Publish to SNS and SQS through App**
$ `curl -H "Content-Type:application/json" -XPOST -d Hello http://localhost:9000/pub/sns`
$ `curl -H "Content-Type:application/json" -XPOST -d Hello http://localhost:9000/pub/sqs`

**See the result**
$ `aws sqs receive-message --queue-url http://localhost:4566/queue/QUEUE-1  --message-attribute-names cargo --max-number-of-messages 2`

```
{
    "Messages": [
        {
            "MessageId": "59e99d3b-7988-44e7-bec5-0901f66c0557",
            "ReceiptHandle": "59e99d3b-7988-44e7-bec5-0901f66c0557#6361781d-767f-49d2-b68e-0069acd4b660",
            "MD5OfBody": "dcc61e5a04e06326ab95f828edb6343b",
            "Body": "{\"size\":92,\"content\":\"Hello\"}",
            "MD5OfMessageAttributes": "ecb2d2e71394aafba89c83d5a36c9257",
            "MessageAttributes": {
                "cargo": {
                    "BinaryValue": "ck8wQUJYTnlBQ0JzYjJOaGJITjBZV05yTG5Ob1lYSmxaQzVVY21GdWMyWmxjazlpYW1WamRIQTFwU090VlJTbEFnQUJUQUFIWTI5dWRHVnVkSFFBRWt4cVlYWmhMMnhoYm1jdlUzUnlhVzVuTzNod2RBQUZTR1ZzYkc4PQ==",
                    "DataType": "Binary"
                }
            }
        },
        {
            "MessageId": "afda700c-e95d-4d94-bc66-b208c97edecc",
            "ReceiptHandle": "afda700c-e95d-4d94-bc66-b208c97edecc#3db3c49c-1201-4a34-af44-86e2ed6c202e",
            "MD5OfBody": "dcc61e5a04e06326ab95f828edb6343b",
            "Body": "{\"size\":92,\"content\":\"Hello\"}",
            "MD5OfMessageAttributes": "53d1936bef4796d2c279268a1b511f11",
            "MessageAttributes": {
                "cargo": {
                    "BinaryValue": "rO0ABXNyACBsb2NhbHN0YWNrLnNoYXJlZC5UcmFuc2Zlck9iamVjdHA1pSOtVRSlAgABTAAHY29udGVudHQAEkxqYXZhL2xhbmcvU3RyaW5nO3hwdAAFSGVsbG8=",
                    "DataType": "Binary"
                }
            }
        }
    ]
}
``` 

You can see that `BinaryValue` values are different, even though the message content that was published was identical.

**Comment by gruff4l0** [comment]
After some investigation, there is a fix. when I build a Docker image and run, both messages arriving as expected.

**Comment by gruff4l0** [comment]
resolved with #2525
